### PR TITLE
MAINT: Implement keyword-only arguments as syntax

### PR DESCRIPTION
**Pull request by eric-wieser**
Now that 2.7 is gone, there is no need to pop manually from kwarg dictionaries.

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -777,7 +777,7 @@ def _gradient_dispatcher(f, *varargs, **kwargs):
 
 
 @array_function_dispatch(_gradient_dispatcher)
-def gradient(f, *varargs, **kwargs):
+def gradient(f, *varargs, axis=None, edge_order=1):
```
Comment (mattip) [inline numpy/lib/function_base.py]: signature mismatch with `_gradient_dispatcher`

**Comment by eric-wieser** [comment]
Numeric einsum failures remain, not sure how I managed to cause those to change.

**Review by seberg — APPROVED** [review]
LGTM, the einsum code is a bit dense/tricky, but that can wait for another time and day. Some functions ate kwargs before, but I _think_ they are not very public (and even if, I am willing to let people run into such an easy to fix error, that was probably a bug).

**Comment by mattip** [comment]
Thanks @eric-wieser

**Comment by eric-wieser** [comment]
> Some functions ate kwargs before

I think I left those out of the patch, or preserved the eating behavior.

**Comment by seberg** [comment]
Yeah, I thought there was one or two that may have changed, but I think those were private. But maybe I misread, in any case, I do not care :).

**Pull request merged**